:sparkles: Add Tasks tab to the Applications table application details drawer

EstyBiton opened the pull request:
Enhancing the application details drawer by adding a "Tasks" tab, displaying a table of tasks associated with the application, implementing filtering, sorting, and pagination functionalities, and including columns for Task ID, Task Kind, Status, and Priority. 

Fixes: #1937 


screenshot of result:
![Screenshot from 2024-10-30 11-36-28](https://github.com/user-attachments/assets/1e36c51d-8586-4c8b-a955-e850a777a4f4)

sjd78 commented on client/src/app/pages/tasks/TaskDetails.tsx:
```diff
@@ -2,27 +2,54 @@ import React from "react";
+  const [applicationName, setApplicationName] = useState<string | undefined>();
+  const [applicationId, setApplicationId] = useState<number | undefined>();
+
+  useEffect(() => {
```
@rszwajko Can you have a look at these changes?  I don't remember exactly how these components hang together.

rszwajko replied:
I missed this comment. I will take a look.

sjd78 reviewed (changes requested):
The work done is quite good.  There are some things that need to be changed.

1. On the side bar table, either drop the priority column or make the initial width of the drawer larger.  With the current width of the drawer, the action kebab are not visible.  The other option is to make the drawer something like 100px wider initially so the task action kebab is shown.

2. When linking to the task details, the URL is breaking.  You can follow the same format as `navigateToAnalysisDetails()`, but it should really be its own path.

3. The status column should use the same column style as the tasks-page (status icon, name, and link).  See: https://github.com/konveyor/tackle2-ui/blob/0cfe60b490e386a360a46e38769b56057845d377/client/src/app/pages/tasks/tasks-page.tsx#L244-L257

4. Need to fix the link that are being generated on task manager table (see specific comment).

I have a few more specific comment in the review.

EstyBiton commented:
@sjd78 Thank you for your review! I’ve made the following updates based on your feedback:
1.   Removed  priority field and unnecessary widths.
2.  Updated rendering: The table now renders like the task manager page, featuring a status icon with a link to view task details.
3. Changed to compact table: I added the variant="compact" prop to enhance the appearance of the table.

rszwajko commented on client/src/app/pages/tasks/TaskDetails.tsx:
Note that after the changes the file duplicates the functionality of `AnalysisDetails` - in fact analysis details is just a task that is highlighted for user convenience. The code behind is/was generic - if you replace the IDs in the URL you should be able to view any task. Ideally you should be able to re-use this code after extracting to some "base" class - similar how both `TaskDetails` and `AnalysisDetails` use `TaskDetailsBase`. IMHO the refactoring (extracting base functionality) is small enough to do in this PR.

It could look the following
```
const  AnalysisDetails = () =>   
  <GoodNameForBaseComponent  
    formatDetailsPath={(applicationId,taskId) => formatPath(Paths.applicationsAnalysisDetails, {
        applicationId,
        taskId,
      });
   formatTitle={(taskName) => `Analysis details for ${taskName}`}
// whatever else is needed
  />  

const  ApplicationTaskDetails = () =>   
  <GoodNameForBaseComponent  
    formatDetailsPath={(applicationId,taskId) => ...);
   formatTitle={(taskName) => ...}
// whatever else is needed
  />  

```

rszwajko commented on client/src/app/pages/applications/components/application-detail-drawer/application-detail-drawer.tsx:
```diff
@@ -521,3 +554,206 @@ const TabReportsContent: React.FC<{
+  const tableControlState = useTableControlState({
+    tableName: "tasks-apps-table",
+    persistTo: "urlParams",
+    persistenceKeyPrefix: TablePersistenceKeyPrefix.tasks,
```
it seems we have a collision with `TasksPage`

rszwajko commented on client/src/app/pages/applications/components/application-detail-drawer/application-detail-drawer.tsx:
```diff
@@ -521,3 +554,206 @@ const TabReportsContent: React.FC<{
   );
 };
+
+const TabTasksContent: React.FC<{
```
Ideally we should be able to reuse the code from `TasksPage`. In the current state this requires a refactoring that would extract the "base" component. The best strategy would be to use a different PR for that and merge it first. This way we (and our test suite) ensure that the refactoring caused no regression.

After the refactoring both `TasksPage` and `TabTasksContent` should use the base and provide the custom parts. There are few differences:
1. identifiers i.e. `tableName`, `persistenceKeyPrefix`
2. visible columns - note that hiding (via `initialColumns`)  looks easier then maintaining 2 lists of columns
3. query params - `implicitFilters`
4.  possibly more

If "base" component variant would be not achievable we can break the TasksPage into smaller parts and try to reuse those.

EstyBiton replied:
@rszwajko Thank you for the feedback! I would really like to continue working on this task, but I currently have a challenge. My internship with the company has concluded, and I no longer have access to the computer where the product environment was properly set up. Unfortunately, running Konveyor on Minikube in the Windows operating system presents a significant issue: the local Konveyor UI is inaccessible.
 If the issue with installing Konveyor is resolved, I’d be happy to continue working on this task. Please let me know.

rszwajko reviewed (commented):
The PR looks pretty impressive! It basically adds the entire feature. We could merge it now however too keep the code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) we need a follow-up refactorings. 
It would be better to merge the main refactoring (extracting base table component) as a prerequisite (different PR), merge it and then use it here. Please let me know if you can work on this (details in the comments). If not we can consider other variants.

sshveta commented:
@rszwajko @sjd78 Can we please review/merge these PRs .

sjd78 reviewed (approved):
Looks ok to me

rszwajko reviewed (approved):
Functionality works as expected according to my smoke tests. The recommended refactorings can be applied in follow-ups.
